04 auth

Veronika-chenko opened the pull request:
Перевір, будь ласка, код 
Із додаткових завдань зробила усі три. Єдине що, замість ,?page=1 робила ?skip=0; Сподіваюся, це ок?

Veronika-chenko committed:
feature/auth

-added registration and login logic
-added token validation middleware

Veronika-chenko committed:
feature/auth

-applied logout logic
-fix router/middlewares/authMiddleware

Veronika-chenko committed:
feature/auth-current-user

added /users/current route to get current user info

Veronika-chenko committed:
fuature/pagination

skip & limit

Veronika-chenko committed:
feature/auth-additions

-filtering contacts by favorite field
-added possibility to update user subscription with patch

Veronika-chenko committed:
feature/auth

KhYulian commented on src/routes/middlewares/authMiddleware.js:
```diff
@@ -0,0 +1,31 @@
+        const [, token] = req.headers.authorization.split(' ')
+
+        if (!token) {
+            res.status(401).json({"message": "0 Not authorized"})
```
return

Veronika-chenko replied:
додала

KhYulian commented on src/routes/middlewares/authMiddleware.js:
```diff
@@ -0,0 +1,31 @@
+
+        if (!token) {
+            res.status(401).json({"message": "0 Not authorized"})
+            next()
```
Якщо авторизація не пройшла успішно next можна викликати хібащо з помилкою. 
Або return res.status(401).json({"message": "0 Not authorized"}) або
return next(new MyError(401, "Not Authorized"))

Veronika-chenko replied:
дякую, так, пропустила цей момент;
прибрала next() і переробила authMiddleware аналогічно до інших ф-цій міддлверів

KhYulian commented on src/routes/middlewares/authMiddleware.js:
```diff
@@ -0,0 +1,31 @@
+        const [, token] = req.headers.authorization.split(' ')
+
+        if (!token) {
+            res.status(401).json({"message": "0 Not authorized"})
```
0?

Veronika-chenko replied:
+

KhYulian commented on src/routes/middlewares/authMiddleware.js:
```diff
@@ -0,0 +1,31 @@
+
+        if (!user || user.token !== token) {
+            res.status(401).json({"message": "Not authorized"})
+            next()
```
Тут аналогічно

Veronika-chenko replied:
прибрала next()

KhYulian commented on src/routes/middlewares/subscriptionMiddleware.js:
```diff
@@ -0,0 +1,13 @@
+const { subscriptionValidation } = require('../../utils/validation/subscriptionValidation')
+
+module.exports = {
+    subscriptionMiddleware: (req, res, next) => {
```
subscriptionValidation

Veronika-chenko replied:
Що тут маєте на увазі?

KhYulian commented on src/services/authService.js:
```diff
@@ -0,0 +1,86 @@
+        return {token, user}
+    } catch (error) {
+        console.log(error)
+    }
```
тут try catch не потрібен. Обробляйте помилки в контроллері

KhYulian replied:
Аналогічно в інших сервісах

Veronika-chenko replied:
прибрала try..catch в усіх сервісах

KhYulian reviewed (changes requested):
Потрібне доопрацювання

KhYulian commented:
> Перевір, будь ласка, код Із додаткових завдань зробила усі три. Єдине що, замість ,?page=1 робила ?skip=0; Сподіваюся, це ок?
Загалом не дуже. skip ви вираховуєте на основі limit і page. skip = (page - 1) * limit;

KhYulian commented:
limit має мати якесь дефолтне значення, А в query має бути page та/або limit

Veronika-chenko committed:
feature/auth

-middleware/authMiddleware.js:
fixed "if" statements with "return" and removed bad next() inside it
-removed try...catch from all services fns
-changed pagination logic with using page and limit from req.query

Veronika-chenko commented:
переробила пагінацію з використанням page i limit з req.query

KhYulian commented:
Не забувайте здавати роботу в лмс)

KhYulian reviewed (approved):
Чудова робота :heart:
